pkg/controller/plan/adapter/ocp/builder.go

```diff
@@ -324,6 +340,144 @@ func (r *Builder) mapDisks(sourceVm *cnv.VirtualMachine, targetVmSpec *cnv.Virtu
 	}
 }
 
+type envMap struct {
+	envResource interface{}
+	volName     string
+}
+
+func (r *Builder) createEnvMaps(sourceVm *cnv.VirtualMachine, vmRef ref.Ref) (map[string]*envMap, map[string]*envMap) {
+	configMaps := make(map[string]*envMap)
+	secrets := make(map[string]*envMap)
+
+	for _, envVol := range sourceVm.Spec.Template.Spec.Volumes {
+		switch {
+		case envVol.ConfigMap != nil:
+			configMap := &core.ConfigMap{}
+			err := r.sourceClient.Get(context.Background(), client.ObjectKey{Namespace: vmRef.Namespace, Name: envVol.ConfigMap.Name}, configMap)
+			if err != nil {
+				r.Log.Error(err, "Failed to get ConfigMap", "namespace", vmRef.Namespace, "name", envVol.ConfigMap.Name)
+				continue
+			}
+			configMaps[envVol.ConfigMap.Name] = &envMap{
+				envResource: configMap,
+				volName:     envVol.Name,
+			}
+
+		case envVol.Secret != nil:
+			secret := &core.Secret{}
+			err := r.sourceClient.Get(context.Background(), client.ObjectKey{Namespace: vmRef.Namespace, Name: envVol.Secret.SecretName}, secret)
+			if err != nil {
+				r.Log.Error(err, "Failed to get Secret", "namespace", vmRef.Namespace, "name", envVol.Secret.SecretName)
+				continue
+			}
+			secrets[envVol.Secret.SecretName] = &envMap{
+				envResource: secret,
+				volName:     envVol.Name,
+			}
+		}
+	}
+
+	return configMaps, secrets
+}
+
+func (r *Builder) mapConfigMapsToTarget(targetVmSpec *cnv.VirtualMachineSpec, configMaps map[string]*envMap, diskMap map[string]*cnv.Disk) {
+	for _, configMap := range configMaps {
+		// Create configmap on destination cluster
+		sourceConfigMap := configMap.envResource.(*core.ConfigMap)
+		targetConfigMap := &core.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        sourceConfigMap.Name,
+				Namespace:   r.Plan.Spec.TargetNamespace,
+				Labels:      sourceConfigMap.Labels,
+				Annotations: sourceConfigMap.Annotations,
+			},
+			Data: sourceConfigMap.Data,
+		}
+		err := r.Destination.Client.Create(context.Background(), targetConfigMap)
+		if err != nil {
+			if !errors.IsAlreadyExists(err) {
+				r.Log.Error(err, "Failed to create ConfigMap", "namespace", r.Plan.Spec.TargetNamespace, "name", targetConfigMap.Name)
+				continue
+			}
+		}
+
+		configMapVolume := cnv.Volume{
+			Name: configMap.volName,
+			VolumeSource: cnv.VolumeSource{
+				ConfigMap: &cnv.ConfigMapVolumeSource{
+					LocalObjectReference: core.LocalObjectReference{
+						Name: targetConfigMap.Name,
+					},
+				},
+			},
+		}
+
+		if disk, ok := diskMap[sourceConfigMap.Name]; ok {
```

Comment:
in which scenario would `ok` be false?

Reply:
hopefully no scenario, but you never know, I wanted to add an else here

Reply:
or a log..

Reply:
yeah that's what I did in the else :)